src/main/java/com/icthh/xm/ms/entity/service/impl/FunctionServiceImpl.java

```diff
@@ -121,6 +123,25 @@ private void validateFunctionInput(FunctionSpec functionSpec, Map<String, Object
         }
     }
 
+    @Override
+    public FunctionContext executeAnonymous(String functionKey, Map<String, Object> functionInput) {
+        FunctionSpec functionSpec = findFunctionSpec(functionKey, null);
+
+        if (functionSpec.getAnonymous() == null || !functionSpec.getAnonymous()) {
+            throw new BusinessException("Not allowed");
+        }
+
+        Objects.requireNonNull(functionKey, "functionKey can't be null");
+        Map<String, Object> vInput = CustomCollectionUtils.emptyIfNull(functionInput);
+
+        dynamicPermissionCheckService.checkContextPermission(DynamicPermissionCheckService.FeatureContext.FUNCTION,
+            FUNCTION_ANONYMOUS_CALL_PRIV, functionKey);
+
+        // execute function
+        Map<String, Object> data = functionExecutorService.execute(functionKey, vInput);
```

Comment:
pls provide separate LEP prefix to ANONYMOUS function

Reply:
added AnonymousFunction prefix